pyro/infer/autoguide/guides.py

```diff
@@ -23,43 +23,14 @@ def model():
 import pyro
```

Comment:
It's now a bit dangerous to `import pyro` from within Pyro. A safer alternative would be to import `primitives`, maybe via something like:
```python
from pyro.primitives import sample as pyro_sample
from pyro.primitives import plate as pyro_plate
```
@eb8680 do you have an opinion?

Reply:
can you please elaborate on where exactly the potential danger lies?

Reply:
Hmm, I guess we can `import pyro` in pyro.infer.

What I was worried about was a cycle like pyro -> pyro.infer -> pyro. But I recall that we decided to restrict `pyro` dependencies to: `poutine`, `logger`, `primitives`, and `util`. Thus it should be fine for `infer` to depend on `pyro`.

Reply:
Resolving, since this seems fine.